multiview/embed/omnibus.py

```diff
@@ -0,0 +1,160 @@
+        details.
+
+    Attributes
+    ----------
+    embeddings_: list of arrays (default = None)
+        List of Omnibus embeddings. One embedding matrix is provided
+        per view. If fit() has not been called, embeddings_ is set to
+        None.
+    """
+
+    def __init__(self, n_components=2, distance_metric="euclidean",
+                 normalize="l1",
+                 algorithm="randomized",
+                 n_iter=5):
+
+        super().__init__()
+        self.n_components = n_components
+        self.normalize = normalize
+        self.distance_metric = distance_metric
+        self.algorithm = algorithm
+        self.n_iter = n_iter
+        self._check_params()
+        self.embeddings_ = None
+
+    def _check_params(self):
+        """
+        Checks that Omnibus arguments are valid. A ValueError
+        is thrown if any are not. The checks performed are:
+            - distance_metric is valid
+            - algorithm is valid
+            - normalize is valid
+            - n_components is positive int
+            - n_iter is positive int
+        """
+
+        valid_metrics = ['braycurtis', 'canberra',
+                         'chebyshev', 'cityblock', 'correlation',
+                         'cosine', 'dice', 'euclidean', 'hamming',
+                         'jaccard', 'jensenshannon', 'kulsinski',
+                         'mahalanobis', 'matching', 'minkowski',
+                         'rogerstanimoto', 'russellrao', 'seuclidean',
+                         'sokalmichener', 'sokalsneath', 'sqeuclidean', 'yule']
+
+        valid_algorithms = ["randomized", "full", "truncated"]
+
+        valid_normalize = ["l1", "l2", "max"]
+
+        if self.distance_metric not in valid_metrics:
+            raise ValueError("distance_metric must be in \
+                             scipy.spatial.distances.pdist.")
+
+        if self.algorithm not in valid_algorithms:
+            raise ValueError("algorithm must be 'randomized', \
+                            'full', or 'truncated'.")
+
+        if self.normalize is not None and \
+           self.normalize not in valid_normalize:
+            raise ValueError("normalize must be 'l2', 'l1', or 'max'.")
+
+        if not isinstance(self.n_components, int) or self.n_components <= 0:
+            raise ValueError("n_components must be positive int.")
+
+        if not isinstance(self.n_iter, int) or self.n_iter <= 0:
+            raise ValueError("n_iter must be positive int.")
+
+    def fit(self, Xs):
+        """
+        Fit the model with Xs and apply the embedding on Xs.
+        The embeddings are saved as a class attribute.
+
+        Parameters
+        ==========
+        Xs: list of array-likes
+            - Xs shape: (n_views,)
+            - Xs[i] shape: (n_samples, n_features_i)
+            The data to embed based on the prior fit function. If
+            view_idx defined, Xs is 2D, single view
```

Comment:
Update the Xs description to be like https://github.com/NeuroDataDesign/multiview/pull/45/files since Xs can also be np.ndarray